proto/api.proto

```diff
@@ -472,6 +472,9 @@ message PushTxnResponse {
 // commit them.
 message ResolveIntentRequest {
   optional RequestHeader header = 1 [(gogoproto.nullable) = false, (gogoproto.embed) = true];
+  // The transaction whose intent is being pushed. The transaction must either
```

Comment:
actually just realized that the comment here is wrong. The Txn can of course be `PENDING` if a txn has been pushed forward in time. Updating.